csaf-import/src/test/kotlin/io/github/csaf/sbom/retrieval/RetrievedProviderTest.kt

```diff
@@ -48,35 +49,41 @@ class RetrievedProviderTest {
 
     @Test
     fun testRetrievedProviderEmptyIndex() = runTest {
-        val loader = CsafLoader(mockEngine())
-        val provider = RetrievedProvider.from("no-distributions.com", loader).getOrThrow()
-        val documentResults = provider.fetchDocuments(loader)
-        assertSame(emptyList(), documentResults)
+        val provider = RetrievedProvider.from("no-distributions.com").getOrThrow()
+        val documentResults = provider.fetchDocuments()
+        assertEquals(0, documentResults.size)
     }
 
     private suspend fun providerTest(url: String) {
-        val loader = CsafLoader(mockEngine())
-        val provider = RetrievedProvider.from(url, loader).getOrThrow()
-        val documentResults = provider.fetchDocuments(loader)
+        val provider = RetrievedProvider.from(url).getOrThrow()
+        val documentResults = provider.fetchDocuments()
         assertEquals(
-            3,
+            4,
             documentResults.size,
-            "Expected exactly 3 results: One document, one document error, one index error"
+            "Expected exactly 4 results: One document, two document errors, one index error"
         )
         // Check some random property on successful document
         assertEquals(
             "Bundesamt für Sicherheit in der Informationstechnik",
             documentResults[0].getOrThrow().json.document.publisher.name
         )
-        // Check document error
+        // Check document validation error
+        val validationException =
+            assertIs<ValidationException>(documentResults[1].exceptionOrNull()?.cause)
+        assertContentEquals(
+            listOf("Filename bsi-2022_2-01.json does not match conformance"),
```

Comment:
this should already be tested by a unit test or are you trying to test the validation error during fetching?

Reply:
Precisely that. If no `ValidationException` is ever raised during fetching, we are missing coverage.